docs/framework/guides/deep-dive/localization.md

```diff
@@ -0,0 +1,186 @@
+
+As the second argument, the translation function accepts a value or an array of values. These "values" will be used to fill the placeholders in the more advanced translation scenarios. And if the `plural` property is specified then the first value will be used as quantity determining the plural form.
+
+When using the `t()` function, you can create your own *localizable messages* or reuse *messages* created in CKEditor 5 packages your project depends on. In case of reusing *messages*, you won't have to worry about translating them as all work will be done by the CKEditor 5 team and Transifex translators anyway (but your help in translating still will be appreciated).
+
+For simple *localizable messages* use the string form for simplicity:
+
+```js
+t( 'insert emoji' );
+t( 'insert %0 emoji', emojiName );
+t( 'insert %0 emoji', [ emojiName ] );
+```
+
+For more advanced scenarios use plain objects forms:
+
+```js
+t( { string: '%0 emoji', id: 'INSERT_EMOJI' }, 'insert' );
+t( { string: '%0 emoji', plural: '%0 emojis', id: 'EMOJI' }, quantity );
+t( { string: '%1 %0 emoji', plural: '%1 %0 emojis', id: 'INSERT_EMOJIS' }, [ quantity, 'Insert' ] );
+```
+
+### Examples
+
+#### Localizing UI of plugins:
+
+```js
+// ...
+editor.ui.componentFactory.add( 'smilingFaceEmoji', locale => {
+	const command = editor.commands.get( 'insertSmilingFaceEmoji' );
+	const buttonView = new ButtonView( locale );
+
+	// The translation function.
+	const { t } = editor.locale;
+
+  // The localized label.
+	const label = t( 'insert smiling face emoji' );
+
+	buttonView.set( {
+		label,
+		icon: emojiIcon,
+		tooltip: true
+	} );
+
+	buttonView.on( 'execute', () => {
+		editor.execute( 'insertSmilingFaceEmoji' );
+		editor.editing.view.focus();
+	} );
+} );
+// ...
+```
+
+Note that this sample lacks a few parts. To check how to create a complete plugin, for example, check the {@link framework/guides/creating-simple-plugin Creating a simple plugin guide}.
+
+#### Localizing aria attributes:
+
+```js
+editingView.change( writer => {
+	const editingView = this._editingView;
+	const t = this.t;
+
+	editingView.change( writer => {
+		const viewRoot = editingView.document.getRoot( this.name );
+
+		writer.setAttribute( 'aria-label', t( 'Rich Text Editor, %0', [ this.name ] ), viewRoot );
+	} );
+} );
+```
+
+#### Localizing pending actions
+
+Pending actions are used to inform the user that the action is in progress and we will lost data while exiting the editor - see {@link module:core/pendingactions~PendingActions the `PendingActions` class}
+
+```js
+_updatePendingAction() {
+	// ...
+	const pendingActions = this.editor.plugins.get( PendingActions );
+
+	// ...
+	const t = this.editor.t;
+	const getMessage = value => `${ t( 'Upload in progress' ) } ${ parseInt( value ) }%.`;
```

Comment:
Could it simply be something like `const getMessage = value => t( 'Upload in progress %0%', parseInt( value ) );`?

I assume the problem is `%0%`? If that's the case maybe we should use a different example or skip it all along, as it doesn't look nice.